pkg/medusa/reconcile.go

```diff
@@ -427,6 +432,125 @@ func UpdateMedusaVolumes(dcConfig *cassandra.DatacenterConfig, medusaSpec *api.M
 			},
 		}
 
-		cassandra.AddOrUpdateVolume(dcConfig, encryptionClientVolume, encryptionClientVolumeIndex, found)
+		newVolumes = append(newVolumes, medusaVolume{
+			Volume:      encryptionClientVolume,
+			VolumeIndex: encryptionClientVolumeIndex,
+			Exists:      found,
+		})
+	}
+	return newVolumes
+}
+
+func MedusaServiceName(clusterName string, dcName string) string {
+	return fmt.Sprintf("%s-%s-medusa-service", clusterName, dcName)
+}
+
+func MedusaStandaloneDeploymentName(clusterName string, dcName string) string {
+	return fmt.Sprintf("%s-%s-medusa-standalone", clusterName, dcName)
+}
+
+func StandaloneMedusaDeployment(medusaContainer *corev1.Container, clusterName, dcName, namespace string, logger logr.Logger) *appsv1.Deployment {
+	medusaStandaloneContainer := *medusaContainer.DeepCopy()
```

Comment:
Issue: if the first thing we do is take a DeepCopy we should probably just be passing a copy, not a pointer.